ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Mempool/Init.hs

```diff
@@ -111,7 +115,30 @@ mkMempool mpEnv = Mempool
     , removeTxs      = implRemoveTxs mpEnv
     , syncWithLedger = fst <$> implSyncWithLedger mpEnv
     , getSnapshot    = snapshotFromIS <$> readTMVar istate
-    , getSnapshotFor = implGetSnapshotFor mpEnv
+    , getSnapshotFor = \slot st ldb (LedgerBackingStoreValueHandle s vh) ->
+         let BackingStoreValueHandle {
+                 bsvhClose
+               , bsvhRangeRead
+               , bsvhRead
+               } = vh
+         in implGetSnapshotFor mpEnv slot st
+             (unExtLedgerStateTables $ changelogDiffs ldb)
+             (LedgerBackingStoreValueHandle s $
+              BackingStoreValueHandle {
+                   bsvhClose
+                 , bsvhRangeRead = \(RangeQuery prev count) ->
+                       fmap unExtLedgerStateTables
+                     . bsvhRangeRead
+                     $ RangeQuery
+                        (fmap ExtLedgerStateTables prev)
+                        count
+                 , bsvhRead =
+                       fmap unExtLedgerStateTables
+                     . bsvhRead
+                     . ExtLedgerStateTables
+                 }
+             )
+
```

Comment:
I think we should inline this in `implGetSnapShotFor`, or make a wrapper around `implGetSnapShotFor` to keep the record here simple